operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/config/ConfigurationService.java

```diff
@@ -362,29 +361,45 @@ default boolean ssaBasedCreateUpdateMatchForDependentResources() {
   }
 
   /**
-   * This is mostly useful as an integration point for downstream projects to be able to reuse the
-   * logic used to determine whether a given {@link KubernetesDependentResource} should use SSA or
-   * not.
-   *
-   * @param dependentResource the {@link KubernetesDependentResource} under consideration
-   * @return {@code true} if SSA should be used for
-   * @param <R> the dependent resource type
-   * @param <P> the primary resource type
-   * @since 4.9.4
```

Comment:
I personally don't think that deprecating should mean removing of old JavaDoc but I don't know what's the custom in this project.

Reply:
I don't think we have one but removing the javadoc makes it even more obvious that people should really be using the other one instead, imo… 😅

Reply:
I've removed the deprecation in the end.